pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -567,6 +582,183 @@ func (v *vpnSeedServer) Deploy(ctx context.Context) error {
 		return err
 	}
 
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, envoyFilter, func() error {
+		envoyFilter.ObjectMeta = metav1.ObjectMeta{
+			Name:      envoyFilter.Name,
+			Namespace: envoyFilter.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				metav1.OwnerReference{
+					APIVersion:         "v1",
+					Kind:               "Namespace",
+					Name:               v.namespace,
+					UID:                v.namespaceUID,
+					Controller:         pointer.BoolPtr(false),
+					BlockOwnerDeletion: pointer.BoolPtr(false),
+				},
+			},
+		}
+		envoyFilter.Spec.WorkloadSelector = &istionetworkingv1alpha3.WorkloadSelector{
+			Labels: v.istioIngressGateway.Labels,
+		}
+		envoyFilter.Spec.ConfigPatches = []*istionetworkingv1alpha3.EnvoyFilter_EnvoyConfigObjectPatch{
```

Comment:
I'm getting dizzy when looking at the nesting here. That's exactly the reason why I went with the YAML approach back as part of #3983. I don't insist on changing this again, but do you want to reconsider this decision? :)

Reply:
We did it the first time as well with the YAML approach however we had problems to fix the tests and then we decided to go this way and do it like in the rest of the file with the `CreateOrUpdate` or call from controllerruntime.